lib3h/src/p2p/p2p_trait.rs

```diff
@@ -0,0 +1,21 @@
+use crate::p2p::p2p_event::P2pEvent;
+use holochain_lib3h_protocol::{DidWork, Lib3hResult};
+
+/// Composition of a Connection, Dht, RuntimeStore, PeerStore, and DataStore
+/// to provide a higher-level Connection interface.
+pub trait P2p {
```

Comment:
If we upgrade `Transport` to look more like [n3h's Connection](https://github.com/holochain/n3h/blob/master/lib/interface/connection-iface.js), then a "P2p" instance really is just an instance that implements both the `Transport` trait and the `Dht` trait perhaps?